src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -417,6 +417,23 @@ template class LogicalExpression<LogicalOperator::AND>;
 template class LogicalExpression<LogicalOperator::OR>;
 
 namespace detail {
+//______________________________________________________________________________
+std::vector<BlockMetadata> evaluatePrefilterExpressionOnMetadata(
+    const std::unique_ptr<PrefilterExpression> prefilterExpr,
+    const std::vector<BlockMetadata>& blocks, const size_t evaluationColumn) {
+  if (blocks.size() <= 2) {
+    return blocks;
+  }
+  BlockMetadata firstBlock = blocks.front();
+  BlockMetadata lastBlock = blocks.back();
+  std::vector<CompressedBlockMetadata> blocksAdjusted(blocks.begin() + 1,
```

Comment:
You can also make the firstBlock and lastBlock `optional` and only delete and reappend them if actually needed
(you have the check lambdas for this already somewhere inside the current `evaluate`.